packages/plugins/response-cache/src/plugin.ts

```diff
@@ -320,90 +327,45 @@ export function useResponseCache<PluginContext extends Record<string, any> = {}>
-                  skip = true;
-                  return;
-                }
-
-                if (scopePerSchemaCoordinate[typename] === 'PRIVATE' && !sessionId) {
-                  skip = true;
-                  return;
-                }
-
-                types.add(typename);
-                if (typename in ttlPerType) {
-                  currentTtl = calculateTtl(ttlPerType[typename], currentTtl);
-                }
 
-                return new Proxy((val: any) => val, {
-                  // Needed for leaf values such as scalars, enums etc
-                  // They don't have fields so visitResult expects functions for those
-                  apply(_, __, [val]) {
-                    return val;
-                  },
-                  get(_, fieldName: string) {
-                    if (fieldName === '__typename') {
-                      typenameCalled++;
-                    }
-                    if (
-                      fieldName === '__leave' &&
-                      /**
-                       * The visitResult function is called for each field in the selection set.
-                       * But visitResult function looks for __typename field visitor even if it is not there in the document
-                       * So it calls __typename field visitor twice if it is also in the selection set.
-                       * That's why we need to count the number of times it is called.
-                       *
-                       * Default call of __typename https://github.com/ardatan/graphql-tools/blob/master/packages/utils/src/visitResult.ts#L277
-                       * Call for the field node https://github.com/ardatan/graphql-tools/blob/master/packages/utils/src/visitResult.ts#L272
-                       */ typenameCalled < 2
-                    ) {
-                      return (root: any) => {
-                        delete root.__typename;
-                        return root;
-                      };
-                    }
-
-                    if (
-                      scopePerSchemaCoordinate[`${typename}.${fieldName}`] === 'PRIVATE' &&
-                      !sessionId
-                    ) {
-                      skip = true;
-                      return;
-                    }
-
-                    if (idFields.includes(fieldName)) {
-                      return (id: string) => {
-                        identifier.set(`${typename}:${id}`, { typename, id });
-                        return id;
-                      };
-                    }
-
-                    return undefined;
-                  },
-                });
-              },
-            },
-          ),
-        );
+      function processResult(data: any) {
+        if (data == null || typeof data !== 'object') {
+          return;
+        }
+        if (Array.isArray(data)) {
+          for (const item of data) {
+            processResult(item);
+          }
+          return;
+        }
```

Comment:
It would be great to find a way to handle this case by adding the typename to the collected entities. This would allow to keep track of cached responses using some types but with empty list or null result.

Today, those response doesn't get invalidated when using `cache.invalidate('Type')`, leading to stale cache hits.

Reply:
But perhaps this can be part of a subsequent PR if you think it's out of scope

Reply:
Just for reference, here is an issue about it (with a test case!) https://github.com/n1ru4l/envelop/issues/1953

Reply:
Let's track that in another PR/issue.